### ISSUE TITLE
Add a way to find reverse dependencies for an indirect dependency

### ISSUE THREAD

**Issue by jjant** [zwilias/elm-json#30]
We've had some indirect dependencies that we didn't know where they came from, it'd be super useful to have something to find all the packages that bring that indirect dependency to our project (from our deps, not the whole elm-package database).

**Comment by brandly** [comment]
i'm sure @zwilias is familiar, but `elm-json tree` is similar in behavior to [`npm ls`](https://docs.npmjs.com/cli/ls). `npm ls` supports an optional argument, where you can specify a package, and it does what you're describing here.

maybe we want to extend `elm-json tree`. currently, it supports an extra argument where you can specify an `elm.json` file.

```shell
$ elm-json help tree
elm-json-tree 
List entire dependency graph as a tree

USAGE:
    elm-json tree [FLAGS] [INPUT]

FLAGS:
    -h, --help       Prints help information
        --test       Promote test-dependencies to top-level dependencies
    -V, --version    Prints version information

ARGS:
    <INPUT>    The elm.json file to solve [default: elm.json]
$ elm-json tree elm/core

-- MISSING ELM.JSON ------------------------------------------------------------

No such file or directory (os error 2)
```

potentially it could check if the value matches `*.json`, and if not, filter the dependencies.

**Comment by zwilias** [comment]
Oh, that's a cool idea @jjant! I'll have a think on how to best handle this, but it should be feasible without too much trouble.

@brandly re the `[INPUT]` stuff: it's similar to how `elm-json install` also supports packagenames and an elm.json path. It'll probably turn into `elm-json tree [FLAGS] [PACKAGE] [-- <INPUT>]`, consistent with other subcommands.

**Comment by zwilias** [comment]
Published in [0.2.8](https://www.npmjs.com/package/elm-json/v/0.2.8)!
